README.md

```diff
@@ -330,6 +330,54 @@ interface Evaluation {
 }
 ```
 
+## Authentication Request Timeout
+
+The `authRequestReadTimeout` option allows you to specify a timeout in milliseconds for the authentication request. If the request takes longer than this timeout, it will be aborted. This is useful for preventing hanging requests due to network issues or slow responses.
+
+If the request is aborted due to this timeout, an ERROR_AUTH event will be emitted, indicating that the authentication failed. Additionally, an ERROR event will be emitted to signal that a general error has occurred.
+
+**This only applies to the authentiaction request. If you wish to set a read timeout on the remaining requests made by the SDK, you may register [API Middleware](#api-middleware)
+
+```typescript
+const options = {
+  authRequestReadTimeout: 30000, // Timeout in milliseconds (default: 30000)
+};
+
+const client = initialize(
+  'YOUR_API_KEY',
+  {
+    identifier: 'Harness1',
+    attributes: {
+      lastUpdated: Date(),
+      host: location.href,
+    },
+  },
+  options
+);
+```
+
+## API Middleware
```

Comment:
Deciding to document this here, mainly to show users that they can use this to timeout the remaining SDK requests like `evaluations` / `metrics`